Update unnamed instances changelog entry (#4122)
purescript/purescript: CHANGELOG.md

  instance Foo Int String
  ```

>   and the compiler will generate a unique name for the instance. Note that
>   generated instance names contain numeric suffixes which can change without
>   warning, and should therefore not be relied on.

Reviewer: As long as we're pedantically revising history, since #4096, the generated names might not have a suffix at all, so even this is technically inaccurate. I might go even further and avoid mentioning any explanation for how generated names can change:
```suggestion
  and the compiler will generate a unique name for the instance. Note that
  generated instance names can change between builds, and should therefore
  not be relied on.
```

Author: This sounds to me like leaving everything else the same, if you compile, delete your output directory, and then compile again, you might get a different generated name, which isn’t true, right?

Author: But #4096 isn’t in any released version so far anyway, correct?

Reviewer: Correct on both counts, but:
1. Do we want to go into that level of detail if the takeaway is ‘don't rely on this’?
2. I think ‘can change without warning’ also allows for the same interpretation (delete and recompile could change it).
3. The revised wording isn't incorrect pre-#4096, just less necessary. I don't know who would be best served by a description of an internal detail that we expect will change in the next release and that we want people to ignore anyway.

Reviewer 2: Can we just change this to
> and the compiler will generate a unique name for the instance. Do not use or rely on the instance name in FFI.

Correct me if I'm wrong, but the above comment was only to clarify that if one were to use it in FFI, it could cause problems because the name could change.

Reviewer: *Something* that draws the line between ‘don't do this’ as a matter of principle and ‘don't do this because’ as a practical warning does seem valuable.

Reviewer 2: Perhaps this?

> and the compiler will generate a unique name for the instance. Do not use or rely on the instance name in FFI as relying on it can produce runtime errors in some situations.

Author: I don't really like that either; it can be irritating to be told that something can cause errors "in some situations" but not be given any clue as to what those situations are.
